frontend/apps/reader/modules/readerlink.lua

```diff
@@ -809,19 +809,25 @@ function ReaderLink:onClearForwardLocationStack()
     return true
 end
 
-function ReaderLink:getPreviousLocationPages()
-    local previous_locations = {}
-    if #self.location_stack > 0 then
-        for num, location in ipairs(self.location_stack) do
-            if self.ui.rolling and location.xpointer then
-                previous_locations[self.document:getPageFromXPointer(location.xpointer)] = num
+function ReaderLink:getLocationPages(is_forward)
+    local location_stack = is_forward and self.forward_location_stack or self.location_stack
```

Comment:
`is_forward` as the parameter name reads odd (the "is" sounds more like it's about an "object", than a "request" as here).
Just `forward_location` and `local location_stack = forward_location and self.forward_location_stack or self.location_stack` would read better.